### PR TITLE
feat(auth): Added OidcProviderConfig type and the Get API

### PR DESCRIPTION
**Pull request by hiranya911**
* Added `Providers` child namespace under `Auth`.
* Added `AuthProviderConfig` and `OidcProviderConfig` types.
* Added `FirebaseAuth.GetOidcProviderConfigAsync()` API and unit tests.

go/firebase-mt-dotnet

Note that original proposal doesn't mention the `Providers` child namespace, but I think it's better to put these types in a child namespace and keep them separate from other more commonly used APIs in Auth. I will push an amendment to the API proposal later.

Merged into idp-config feature branch.

### CODE REVIEW

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+namespace FirebaseAdmin.Auth.Providers.Tests
+{
+    public class ProviderConfigManagerTest
+    {
+        private const string MockProjectId = "project1";
+
+        private const string ClientVersionHeader = "X-Client-Version";
+
+        private const string IdTooklitUrl = "https://identitytoolkit.googleapis.com/v2/projects/{0}";
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs:32]: `IdToolkitUrl`?

Comment (MathBunny) [reply FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs:32]: Actually I think this isn't used in this test

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/ProviderConfigManager.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    /// ProviderConfigManager provides methods for interacting with the
+    /// <a href="https://developers.google.com/identity/toolkit/web/reference/relyingparty">
+    /// Google Identity Toolkit v2</a> via its REST API. This class does not hold any mutable
+    /// state, and is thread safe.
+    /// </summary>
+    internal sealed class ProviderConfigManager : IDisposable
+    {
+        private const string ClientVersionHeader = "X-Client-Version";
+
+        private const string IdTooklitUrl = "https://identitytoolkit.googleapis.com/v2/projects/{0}";
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin/Auth/Providers/ProviderConfigManager.cs:35]: I think there's a typo and you meant `IdToolkitUrl`

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            ""issuer"": ""https://oidc.com/issuer"",
+            ""displayName"": ""oidcProviderName"",
+            ""enabled"": true
+        }";
+
+        private static readonly string ConfigNotFoundResponse = @"{
+            ""error"": {
+                ""message"": ""CONFIGURATION_NOT_FOUND""
+            }
+        }";
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs:51]: Not sure, can these be `const` as well?

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+
+namespace FirebaseAdmin.Auth.Providers
+{
+    /// <summary>
+    /// Represents an OIDC auth provider configuration. See
+    /// https://openid.net/specs/openid-connect-core-1_0-final.html.
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs:21]: Maybe having a hyperlink would be nice here

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        /// OIDC provider's ID token signature.</description>
+        /// </item>
+        /// <item>
+        /// <description>To determine the <c>claims_supported</c> to construct the user attributes
+        /// to be returned in the additional user info response.</description>
+        /// </item>
+        /// </list>
+        /// <para>
+        /// ID token validation is performed as defined in
+        /// https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation.
```
Comment (MathBunny) [inline FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs:60]: Maybe a hyperlink can be used here (if you think it's helpful)

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs (hunk 1 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs:32]: Removed.

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/ProviderConfigManager.cs (hunk 2 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/Providers/ProviderConfigManager.cs]: Fixed. This was copied over from `FirebaseUserManager`. I've fixed that instance too. Good catch!

File: FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs (hunk 3 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/ProviderConfigManagerTest.cs]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs (hunk 4 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs (hunk 5 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/Providers/OidcProviderConfig.cs]: Done

**Pull request merged**